### ISSUE TITLE
Add error code for docstring styles that deviate from selected style

### ISSUE THREAD

**Issue by alpastor4** [jsh9/pydoclint#169]
If a user is linting sphinx-style docstrings, pydoclint should be able to detect whether docstrings deviating from the selected style, such as google/numpy, are present and raise a violation.

**Comment by jsh9** [comment]
Hi @alpastor4 , @kehoecj, @jack-swiney:

If I were to implement this feature, changes would need to be made here: https://github.com/jsh9/pydoclint/blob/27ff76915d548ebc19bcd1e9f802bb7c0386b4a6/pydoclint/utils/doc.py#L26-L35

The logic would be:

1. Parse the docstring using 3 different parsers
2. Write some logic to determine what style the docstring actually is
3. Report violation if there's discrepancy

The difficulty is in Step 2, because the logic isn't trivial to write, especially when the dosctring is written with a mixed style (I don't know how often it happens).

Would you want to try to write some logic yourself?  If it is solid, I can consider implement it.

Here's a script that you can use as a basis for experimenting the logic:

```python
# Install first: https://github.com/jsh9/docstring_parser_fork
from docstring_parser.google import GoogleParser
from docstring_parser.numpydoc import NumpydocParser
from docstring_parser.rest import parse as parseSphinx


docstring = """Fetches rows from a Smalltable.

    Retrieves rows pertaining to the given keys from the Table instance
    represented by table_handle.  String keys will be UTF-8 encoded.

    Args:
        table_handle: An open smalltable.Table instance.
        keys: A sequence of strings representing the key of each table
          row to fetch.  String keys will be UTF-8 encoded.
        require_all_keys: If True only rows with values set for all keys will be
          returned.

    Returns:
        A dict mapping keys to the corresponding table row data
        fetched. Each row is represented as a tuple of strings. For
        example:

        {b'Serak': ('Rigel VII', 'Preparer'),
         b'Zim': ('Irk', 'Invader'),
         b'Lrrr': ('Omicron Persei 8', 'Emperor')}

        Returned keys are always bytes.  If a key from the keys argument is
        missing from the dictionary, then that row was not found in the
        table (and require_all_keys must have been False).

    Raises:
        IOError: An error occurred accessing the smalltable.
"""

numpy_parser = NumpydocParser()
parsed_numpy = numpy_parser.parse(docstring)

google_parser = GoogleParser()
parsed_google = google_parser.parse(docstring)

parsed_sphinx = parseSphinx(docstring)

print(parsed_numpy)
print('--------------------')
print(parsed_google)
print('--------------------')
print(parsed_sphinx)

```

**Comment by Gabriel-p** [comment]
How about just showing a warning/error when something was not found? For example: `function xxx.py: WARNING: No Long description found in docstring`?

And if the parser founds nothing (empty or really badly formatted docstring) raise a `function xxx.py: Could not parse docstring`?

**Comment by jsh9** [comment]
Hi @Gabriel-p , sorry for the late reply!

> How about just showing a warning/error when something was not found? For example: `function xxx.py: WARNING: No Long description found in docstring`?

Lack of long description is entirely fine in a docstring, and also it's out of the scope of _pydoclint_.  Because _pydoclint_'s goal is to check consistency between docstring args/exceptions/returns against args/exceptions/returns in the function's signature or body.

> And if the parser founds nothing (empty or really badly formatted docstring) raise a `function xxx.py: Could not parse docstring`?

I could potentially implement this idea, but this doesn't solve @alpastor4's original issue, which is to detect mixture of docstring styles.

I think a more elegant way is to write a new linter that contains a machine learning model to detect docstring styles.  It will essentially be a 4-category classification model:
* Category 0: unrecognizable or mixed
* Category 1: numpy
* Category 2: google
* Category 3: sphinx

Such a machine learning model should be very easy to train and with very high accuracy (although not 100%).

But I don't know how much demand are there in the community, and I don't know how to gather such data.  A poll or something?  I'm not sure.

**Comment by Gabriel-p** [comment]
I think that my issue is simpler than the one reported by the OP. To me, having a warning shown when the docstring could not be parsed with the selected style would be enough. I don't need the package to auto-detect the docstring's style, I just need it to warn me if my docstring does not conform to the style I selected.

Should I open a new issue?

**Comment by jsh9** [comment]
> I think that my issue is simpler than the one reported by the OP. To me, having a warning shown when the docstring could not be parsed with the selected style would be enough. I don't need the package to auto-detect the docstring's style, I just need it to warn me if my docstring does not conform to the style I selected.
> 
> Should I open a new issue?

Sure, please open a new issue. Thanks!

**Comment by jsh9** [comment]
Hi @alpastor4 , @kehoecj, @jack-swiney,

I've published a new version of pydoclint (0.6.0) to add this check.

You can read more about this new feature here: https://jsh9.github.io/pydoclint/style_mismatch.html
